pre-push: unmapped source should skip local pytest instead of fallback full suite

## 背景
当前 pre-push 增量测试选择器在出现任意 unmapped source 时，会直接回退到全量 `tests/vibe3`，本地 push 变慢（常见 2 分钟+）。

本次现场日志示例：
- `Running test suite (full): source-to-test mapping missing, fallback to full suite`

## 现状代码
- 选择器：`src/vibe3/services/pre_push_test_selector.py`
  - `unmapped_sources` 非空时返回：
    - `mode="full"`
    - `tests=["tests/vibe3"]`
    - `reason="source-to-test mapping missing, fallback to full suite"`
- 执行入口：`scripts/hooks/pre-push.sh`
  - 无论 mode 如何，都会执行 `uv run pytest "${TEST_TARGETS[@]}"`

## 期望
当映射缺失时，不在 pre-push 本地做全量 pytest，改为：
1. 输出明确 warning（列出 unmapped sources）
2. 本地 pre-push 跳过 pytest（或仅极小 smoke，优先跳过）
3. 把完整验证交给 CI

## 建议实现
- `PrePushTestSelection.mode` 增加 `skip`（或复用 `smoke` + 空 targets 语义）
- `pre-push.sh` 对 `mode=skip` 直接打印说明并继续后续风险评估步骤
- 保留 `VIBE_PREPUSH_FULL=1` 强制全量能力

## 验收标准
- 修改一个未映射的 `src/vibe3/*.py` 时，pre-push 不再跑全量 `tests/vibe3`
- 终端输出可见：哪些文件未映射、为何跳过本地 pytest
- CI 仍按现有规则跑完整检查
- 增补/更新对应测试：
  - `tests/vibe3/services/test_pre_push_test_selector.py`
  - `tests/vibe3/integration/test_review_shell_contract.py`

### [Manager] Issue Claimed

- **State Transition:** state/ready -> state/claimed
- **Scene Audit:** Healthy (worktree and branch intact at /Users/jacobcy/src/vibe-center/main/.worktrees/task/issue-303).
- **Worktree Status:** Local uncommitted changes exist for 'scripts/hooks/pre-push.sh' and related tests.
- **Findings:**
    - The issue was resumed from state/ready per the latest human instruction.
    - Preliminary verification shows one test failure in 'tests/vibe3/analysis/test_pre_push_test_selector.py' (AssertionError: incremental != smoke).
    - The implementation of 'mode="skip"' and unmapped sources listing is partially present in the working directory but requires verification and commit.
    - System-level 'Flow not found' error encountered in vibe3 CLI, though handoff append functionality is operational.
- **Manager Next Step:** Handing over to the **Plan Agent** to verify the implementation strategy, resolve the regression, and prepare for execution.

### [Manager] Implementation Review & Cleanup

• **State Transition**: state/handoff -> state/in-progress
• **Scene Audit**: Healthy (Worktree /Users/jacobcy/src/vibe-center/main/.worktrees/task/issue-303)
• **Findings**:
  1. **Implementation Verified**: The core requirement to display unmapped sources in `pre-push.sh` and support `mode="skip"` is implemented and verified in the execution report.
  2. **Test Regression**: `tests/vibe3/analysis/test_pre_push_test_selector.py` has one failure (`AssertionError: incremental != smoke`). This is a test expectation mismatch: by adding the missing `test_review_shell_contract.py`, hook changes now resolve to incremental tests instead of falling back to smoke.
  3. **Uncommitted Changes**: All implementation and test changes are currently uncommitted in the worktree.
  4. **Tool Status**: `vibe3` CLI is currently reporting 'Flow not found' for this branch, although handoff operations remain functional.

• **Manager Next Step**: 
  - Handing back to the **Executor** to fix the test expectation in `test_pre_push_test_selector.py`.
  - Once tests pass, the Executor must commit the changes and provide an updated `report_ref`.
  - Review will be skipped for the fix itself, but a full audit will follow once the PR is ready.

--- a/src/vibe3/analysis/pre_push_test_selector.py
+++ b/src/vibe3/analysis/pre_push_test_selector.py
@@ -81,6 +81,19 @@ def select_pre_push_tests(
             test_dir = root / "tests" / "vibe3" / src_rel.parent
             if test_dir.exists() and any(test_dir.glob("test_*.py")):
                 dir_targets.add(test_dir.relative_to(root).as_posix())
+        # Guard: if any resolved directory is the full test suite root,
+        # skip local pytest instead of running everything.
+        if "tests/vibe3" in dir_targets:
+            return PrePushTestSelection(
+                mode="skip",
+                tests=[],
+                reason=(
+                    f"unmapped source(s) resolve to full test suite root, "
+                    f"skipping local run (CI covers full suite): "
+                    f"{', '.join(sorted(unmapped_sources))}"
+                ),
+                unmapped_sources=sorted(unmapped_sources),
+            )
         if dir_targets:
             return PrePushTestSelection(
                 mode="incremental",

--- a/tests/vibe3/analysis/test_pre_push_test_selector.py
+++ b/tests/vibe3/analysis/test_pre_push_test_selector.py
@@ -80,3 +80,15 @@ def test_maps_hook_changes_to_hook_regression_tests() -> None:
 
     assert selection.mode == "smoke"
     assert selection.tests == ["tests/vibe3/analysis/test_pre_push_scope.py"]
+
+
+def test_top_level_unmapped_source_skips_full_suite() -> None:
+    # src/vibe3/__init__.py has no direct test file mapping.
+    # Its directory fallback would resolve to tests/vibe3 (the full suite root).
+    # Must return mode=skip instead of running everything locally.
+    selection = select_pre_push_tests(["src/vibe3/__init__.py"])
+
+    assert selection.mode == "skip"
+    assert selection.tests == []
+    assert selection.unmapped_sources == ["src/vibe3/__init__.py"]
+    assert "tests/vibe3" not in selection.tests

